spec/app/domain/authentication/authn_jenkins/authenticator_spec.rb

```diff
@@ -0,0 +1,120 @@
+    double('HTTPResponse', 
+            code: 200, 
+            body: '{"_class":"org.jenkinsci.plugins.workflow.job.WorkflowRun","actions":[{"_class":"hudson.model.CauseAction","causes":[{"_class":"hudson.model.Cause$UserIdCause","shortDescription":"Started by user Admin","userId":"admin","userName":"Admin"}]},{},{},{},{},{},{"_class":"org.jenkinsci.plugins.pipeline.modeldefinition.actions.RestartDeclarativePipelineAction"},{},{},{},{"_class":"org.jenkinsci.plugins.workflow.job.views.FlowGraphAction"},{},{},{}],"artifacts":[],"building":true,"description":null,"displayName":"#15","duration":0,"estimatedDuration":372,"executor":{"_class":"hudson.model.OneOffExecutor"},"fullDisplayName":"testJob #15","id":"15","keepLog":false,"number":15,"queueId":104,"result":null,"timestamp":1571239175624,"url":"http://10.32.64.101:8080/job/testJob/15/","changeSets":[],"culprits":[],"nextBuild":null,"previousBuild":{"number":14,"url":"http://10.32.64.101:8080/job/testJob/14/"}}'
+    )
+  end
+
+  def invalid_build_response_build_false
+    double('HTTPResponse', 
+            code: 200,
+            body: '{"_class":"org.jenkinsci.plugins.workflow.job.WorkflowRun","actions":[{"_class":"hudson.model.CauseAction","causes":[{"_class":"hudson.model.Cause$UserIdCause","shortDescription":"Started by user Admin","userId":"admin","userName":"Admin"}]},{},{},{},{},{},{"_class":"org.jenkinsci.plugins.pipeline.modeldefinition.actions.RestartDeclarativePipelineAction"},{},{},{},{"_class":"org.jenkinsci.plugins.workflow.job.views.FlowGraphAction"},{},{},{}],"artifacts":[],"building":false,"description":null,"displayName":"#15","duration":0,"estimatedDuration":372,"executor":{"_class":"hudson.model.OneOffExecutor"},"fullDisplayName":"testJob #15","id":"15","keepLog":false,"number":15,"queueId":104,"result":null,"timestamp":1571239175624,"url":"http://10.32.64.101:8080/job/testJob/15/","changeSets":[],"culprits":[],"nextBuild":null,"previousBuild":{"number":14,"url":"http://10.32.64.101:8080/job/testJob/14/"}}'
+    )
+  end
+
+  def invalid_build_response_not_200
+    double('HTTPResponse', 
+            code: 400,
+            body: ''
+    )
+  end
+
+  def invalid_public_key
+    double('HTTPResponse', 
+            code: 200,
+            body: '',
+            headers: {'X-Instance-Identity' => 'notreal'}
+    )
+  end
+
+  def invalid_public_key_not_available
+    double('HTTPResponse', 
+            code: 200,
+            body: ''
+    )
+  end
+
+  def valid_public_key
+    double('HTTPResponse', 
+            code: 200,
+            body: 'MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAhU/Sg1ubdTES9qVSj45US8SRKh3PCvXwezMzJ8Xp7Rhi0qpPQmSKireaJ+R7+yUkY9xsZ7eTxFw2gCJz5fQyWoryjwJaA9ZsDU6jFZIv+SYHvTf3LTZ+TWeYi6A/gxF6JMckawgviQ1MV9VuBan5D8B/P4GR7TbmqiZIvBWfjSayz3Yj+85/PraBJJNC8lTYK61XjDd981nGNddcJ1KG1ZUarsBKXEX45cNEQJ21ZpisELVpXySk4czOSA52bADsaLJDX5MKNjbQgG+jd9tWo+8w3J7rKPThLRroMGukqMT7l535YnwQ0IocbvrM5uX3okydwcEADNYr8QCW1BhxUwIDAQAB'
+    )
+  end
+
+  def valid_login
```

Comment:
Is this method used anywhere?  If not, we should get rid of it.

Reply:
Nope they are not used and can be removed. Resolved in this commit: https://github.com/AndrewCopeland/conjur/commit/9badd0d64ef96774aa946dc1add0e6e6bd6dc0ec